sklearn/metrics/pairwise.py

```diff
@@ -2000,7 +2001,7 @@ def _pairwise_callable(X, Y, metric, ensure_all_finite=True, **kwds):
 
     else:
         # Calculate all cells
-        out = np.empty((X.shape[0], Y.shape[0]), dtype="float")
+        out = xp.empty((X.shape[0], Y.shape[0]), dtype="float")
```

Comment:
```suggestion
        out = xp.empty((X.shape[0], Y.shape[0]), dtype=xp.float64)
```

would be equivalent to the previous numpy behavior.

However it might fail for some namespace and device combinations that might not support `xp.float64`. For instance torch with the "mps" device only support `xp.float32`.

So it would probably be safer to allocate out with same dtype as `X`:

```suggestion
        out = xp.empty((X.shape[0], Y.shape[0]), dtype=X.dtype)
```

This is a slight change of behavior though if `X` is a numpy array with an `np.float32` dtype... but maybe it might be considered as a bugfix?